doc/api/stream.md

```diff
@@ -1359,6 +1364,21 @@ The `writable._writev()` method is prefixed with an underscore because it is
 internal to the class that defines it, and should never be called directly by
 user programs.
 
+#### writable.\_final(callback)
+<!-- YAML
+added: REPLACEME
+-->
+
+
+* `callback` {Function} Call this function (optionally with an error
+  argument) when you are done writing any remaining data.
+
+Note: This function **must not** be called directly.  It MAY be implemented
+by child classes, and if so, will be called by the internal Writable
+class methods only.
+
+This optional function will be called before the stream closes, delaying the `finish` event until `callback` is called. This is useful to close resources or write buffered data before a stream ends.
```

Comment:
Thank you. This LGTM. If CI is green and @nodejs/streams signs off it should be good to go as far as I'm concerned.